x-pack/plugin/core/src/main/resources/ecs-template.yml

```diff
@@ -0,0 +1,225 @@
+      # No path matching is needed as it is assumed these fields are already in the json doc a long
+      # Elasticsearch would do this mapping automatically but it is specified here for documentation purpose
+      # DISCUSS: If a field "7" as string comes in, will it still be mapped correctly?
+      - long_as_long:
+          match_mapping_type: long
+          runtime:
+            type: long
+
+      # Matching all long and double fields
+      # This matches the following fields: event.risk_score
+      # No path matching is needed as it is assumed these fields are already in the json doc a float or double
+      # Elasticsearch would do this mapping automatically but it is specified here for documentation purpose
+      # DISCUSS: If a field "7.7" as string comes in, will it still be mapped correctly?
+      - float_as_double:
+          match_mapping_type: float
+          runtime:
+            type: double
+      - double_as_double:
+          match_mapping_type: double
+          runtime:
+            type: double
+
+      # Matching all boolean fields
+      # This matches the following fields: exists
+      # No path matching is needed as it is assumed these fields are already in the json doc as boolean
+      # Elasticsearch would do this mapping automatically but it is specified here for documentation purpose
+      # DISCUSS: If a field "true" as string comes in, will it still be mapped correctly?
+      - strings_as_boolean:
+          match_mapping_type: boolean
+          runtime:
+            type: boolean
+
+      # Matching all string fields by default to keyword
+      # TODO: Does first / last match win in the list?
+      - strings_as_keyword:
+          match_mapping_type: string
+          runtime:
+            type: keyword
+            # Do we need this for runtime?
+            ignore_above: 1024
+
+      # Any message field except the top level one is matched as text but not indexed
+      # TODO: Does this work? Which one has priority?
+      - match_message:
+          match_mapping_type: string
+          match: message
+          runtime:
+            type: match_only_text
+
+      # Match all ip fields to ip
+      - match_ip:
+          match_mapping_type: string
+          # DISCUSS: If match could be an array, these mappings could be written much more compact
+          match: ip
+          runtime:
+            type: ip
+
+      # Match all forwarded_ip fields to ip. This is for network.forwarded_ip
+      - match_ip:
+          match_mapping_type: string
+          match: forwarded_ip
+          runtime:
+            type: ip
+
+      # Match all labels fields. Flattened type instead of keyword is used to prevent too many fields to be created.
+      # DISCUSS: Is this the right choice?
+      - match_labels:
+          match_mapping_type: string
+          # TODO: Does this also match the top level labels?
+          match: labels
+          runtime:
+            type: flattened
```

Comment:
Runtime fields don't support flattened.

See https://www.elastic.co/guide/en/elasticsearch/reference/current/runtime-mapping-fields.html for a list of supported field types.

Just relying on dynamic: runtime seems sufficient for labels. But if we want to index them, either flattened or subobjects: false may be a good choice.

Creating too many fields doesn't seem unique to labels, there may also be many top-level fields.

Reply:
Labels by nature have a lot of fields as these are for example k8s labels and not fields that are directly user created. 

@jsoriano I remember you were involved by the time flattened was introduced but it seems we don't use it for labels today but some other places. Did we just never switch over?

Reply:
We had this [private] https://github.com/elastic/obs-infraobs-team/issues/461 conversation. The general output was to wait till there is better general support, and evaluate case per case. Now using `subobjects` may be also a good choice, except on places where we are also worried about fields explosion.

Reply:
if we do want to use flattened, these would be indexed though, not runtime. How often are these labels queried / aggregated on?

Reply:
I expect in a k8s use case, filter queries on these could happen frequently, not sure about aggregations. But I would prefer to not have these indexed by default but also allow it in runtime.

Is the following correct: If the field is runtime, keyword vs flattened does not really make a difference. Only when indexed there is a difference between flattened and keyword. So basically as long as it is a runtime, we keep it keyword, only for index we recommend flattened. If yes, this leaves us still with the topic that the number of fields could explode because of lots of labels but this is something that could be overcome by the user based on the setting or with the idea from @felixbarny to have a soft limit for # of fields.

Reply:
> If the field is runtime, keyword vs flattened does not really make a difference.

Runtime fields don't support flattened. Flattened implies a specific way of indexing the fields. If we want to use runtime fields for labels, the best option is to just not map them explicitly and rely on dynamic: runtime.